pkg/operator2/routercerts/controller.go

```diff
@@ -0,0 +1,251 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	controllerWorkQueueKey           = "key"
+	conditionRouterCertsDegradedType = "RouterCertsDegraded"
+)
+
+// RouterCertsDomainValidationController validates that router certs match the ingress domain
+type RouterCertsDomainValidationController struct {
+	operatorClient v1helpers.OperatorClient
+	cachesToSync   []cache.InformerSynced
+	queue          workqueue.RateLimitingInterface
+	eventRecorder  events.Recorder
+
+	ingressLister configv1listers.IngressLister
+	secretLister  corev1listers.SecretNamespaceLister
+	secretName    string
+	routeName     string
+}
+
+func NewRouterCertsDomainValidationController(
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+	ingressLister configv1listers.IngressLister,
+	secretLister corev1listers.SecretNamespaceLister,
+	secretName string,
+	routerName string,
+
+) *RouterCertsDomainValidationController {
+	controller := &RouterCertsDomainValidationController{
+		operatorClient: operatorClient,
+		cachesToSync:   nil,
+		queue:          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "RouterCertsDomainValidationController"),
+		eventRecorder:  eventRecorder,
+		ingressLister:  ingressLister,
+		secretLister:   secretLister,
+		secretName:     secretName,
+		routeName:      routerName,
+	}
+	operatorClient.Informer().AddEventHandler(controller.newEventHandler())
+	controller.cachesToSync = append(controller.cachesToSync, operatorClient.Informer().HasSynced)
+	return controller
+}
+
+func (c *RouterCertsDomainValidationController) sync() error {
+	spec, _, _, err := c.operatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	if !management.IsOperatorManaged(spec.ManagementState) {
+		return nil
+	}
+
+	condition, err := c.validateRouterCertificates()
+	if err != nil {
+		return err
+	}
+
+	condition.Type = conditionRouterCertsDegradedType
+	_, _, err = v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(*condition))
+	return err
+}
+
+func (c *RouterCertsDomainValidationController) validateRouterCertificates() (*operatorv1.OperatorCondition, error) {
+	// get ingress
+	ingress, err := c.ingressLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		return &operatorv1.OperatorCondition{
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "NoIngressConfig",
+			Message: fmt.Sprintf("ingresses.config.openshift.io/cluster: %v", err),
+		}, nil
+	}
+	if err != nil {
+		return nil, err
```

Comment:
This doesn't make sense.  If you need to be degraded, be degraded.  I think all errors here are uniform.

Reply:
Done